File: leetcode-go/Q001-Q100/Q080_Remove_Duplicates_from_Sorted_Array_II/Q080_Remove_Duplicates_from_Sorted_Array_II.go
package Q080_Remove_Duplicates_from_Sorted_Array_II

func removeDuplicates(nums []int) int {
	if len(nums) < 3 {
		return len(nums)
	}
	idx := 1
	for i := 2; i < len(nums); i++ {
		if !(nums[i] == nums[idx] && nums[i] == nums[idx-1]) {
			idx++
			nums[idx] = nums[i]
		}
	}
	return idx
}

File: leetcode-java/src/facebook/Q080_Remove_Duplicates_from_Sorted_Array_II.java
package facebook;

public class Q080_Remove_Duplicates_from_Sorted_Array_II {

    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        int k = 0, cnt = 0;
        for (int i = 0; i < nums.length; i++) {
            if (k == 0) {
                cnt = 1;
                nums[k++] = nums[i];
            } else if (nums[i] != nums[i-1]) {
                cnt = 1;
                nums[k++] = nums[i];
            } else if (nums[i] == nums[i-1]){
                cnt++;
                if (cnt <= 2) {
                    nums[k++] = nums[i];
                }
            }
        }
        return k;
    }

}
